### PR TITLE
Create IDEs and Editors 2019 roadmap

### PR DESCRIPTION
**Pull request by Xanewok**
cc @Manishearth @nrc @vlad20012 @matklad

### CODE REVIEW

**Review by vlad20012 — APPROVED** [review]

File: roadmaps/ides-and-editors-2019.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## IntelliJ Rust
+
+**Code analysis** is big on our agenda for this year. As you might already know, we don’t use RLS/Racer or rustc whatsoever in IntelliJRust. Instead, our lexing, parsing, type inference, and all the code analyzing stuff is written from the ground up. It lets us set up prominent code completion, on-demand code analysis, and have low latency even on a large codebase. The drawback of this is that from time to time it may behave a little differently to rustc. Typical examples of bad behavior include false positives, some names are unresolved, or some types may not infer.  
+This year this situation is going to change drastically. We’re making our on-the-fly analysis operation as close to rustc, as possible. The plan is to put together an analysis testing system for real-life projects (somewhat similar to Crater), that will give us an indication of what the most common issues from real projects are for us to prioritize. This should help us to make our code analysis features gradually better over the course of the year.
```
Comment (Manishearth) [inline roadmaps/ides-and-editors-2019.md]: nit: "as close to rustc as possible", no comma

**Review by Manishearth — APPROVED** [review]
This looks great!

We'll probably want a TLDR for the whole-team roadmap when we make one, but no rush on that.

File: roadmaps/ides-and-editors-2019.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# IDEs and editors - 2019 planning
+Looking back at 2018, our team managed to ship RLS (Rust language server) out of preview state with the Rust 2018 edition release, spawn a separate IDE project dubbed Rust Analyzer that’s focused on latency and lazy computation, prepare Racer for the Rust 2018 edition, and more!
+
+We’re already more than a little into 2019 so now it’s a good time to plan ahead and see what this year may look like for IDEs and editors! Below you can find what we’ll have in store for the relevant tools.
+
+## RLS 2.0 (…?)
+
+Fresh out of Rust All Hands, which was held and attended by the Rust team in Berlin two weeks ago (Feb 2019), we’re excited to say that we spent a considerate amount of time talking about the future of the Rust IDE support and the fruit of that is formation of an experimental RLS 2.0 compiler working group:
+
+https://github.com/rust-analyzer/WG-rls2.0
```
Comment (matklad) [inline roadmaps/ides-and-editors-2019.md]:
```suggestion
https://github.com/rust-lang/compiler-team/tree/master/working-groups/rls-2.0
```

We've moved :)

**Comment by Xanewok** [comment]
Fixed comments.

**Pull request merged**